Integration of new fota framework with subdevice. (#59)
PelionIoT/mbed-edge: README.md

After starting Edge Core, you can start the protocol translator. It connects
then to Edge Core:

> ## Configuring the Subdevice Fota

Reviewer: ```suggestion
### Configuring the subdevice FOTA
```

> To enable the subdevice fota, you just need to build the edge-core with the FOTA_ENABLE option in cmake. You can also configure the download location of the firmware.

Reviewer: @prakharbhatt007 whats the default location if not specified?

> To enable the subdevice fota, you just need to build the edge-core with the FOTA_ENABLE option in cmake. You can also configure the download location of the firmware.

Reviewer: ```suggestion
To enable the subdevice FOTA, you need to build edge-core with the `FOTA_ENABLE` along with `FIRMWARE_UPDATE` cmake flag. You can also configure the download location of the firmware by explicitly defining the `SUBDEVICE_FIRMWARE_DOWNLOAD_LOCATION` flag. By default the firmware will be downloaded to the working directory. 
```

For example:
``` bash

>     cmake -D[your build options]  -DSUBDEVICE_FIRMWARE_DOWNLOAD_LOCATION=\"your_download_location\" ..

Reviewer: ```suggestion
    cmake -D[mode] -DFIRMWARE_UPDATE -DFOTA_ENABLE  -DSUBDEVICE_FIRMWARE_DOWNLOAD_LOCATION=\"your_download_location\" ..
```
